csharp/Microsoft.Azure.Databricks.Client/PipelinesApiClient.cs

```diff
@@ -0,0 +1,223 @@
+        int? expectedLastModified = null,
+        CancellationToken cancellationToken = default)
+    {
+        var requestUri = $"{ApiVersion}/pipelines/{pipelineId}";
+        var request = JsonSerializer.SerializeToNode(pipelineSpecification, Options)!.AsObject();
+        request.Add("pipeline_id", pipelineId);
+        request.Add("allow_duplicate_names", allowDuplicateNames);
+
+        if (expectedLastModified != null)
+        {
+            request.Add("expected_last_modified", expectedLastModified.Value);
+        }
+
+        await HttpPut<JsonObject>(this.HttpClient, requestUri, request, cancellationToken).ConfigureAwait(false);
+    }
+
+    public async Task Delete(string pipelineId, CancellationToken cancellationToken = default)
+    {
+        var requestUri = $"{ApiVersion}/pipelines/{pipelineId}";
+        await HttpDelete(this.HttpClient, requestUri, cancellationToken).ConfigureAwait(false);
+    }
+
+    public async Task Reset(string pipelineId, CancellationToken cancellationToken = default)
+    {
+        var requestUri = $"{ApiVersion}/pipelines/{pipelineId}/reset";
+        await HttpPost(this.HttpClient, requestUri, cancellationToken);
+    }
+
+    public async Task Stop(string pipelineId, CancellationToken cancellationToken = default)
+    {
+        var requestUri = $"{ApiVersion}/pipelines/{pipelineId}/stop";
+        await HttpPost(this.HttpClient, requestUri, cancellationToken);
+    }
+
+    public async Task<Pipeline> Get(string pipelineId, CancellationToken cancellationToken = default)
+    {
+        var requestUri = $"{ApiVersion}/pipelines/{pipelineId}";
+        return await HttpGet<Pipeline>(this.HttpClient, requestUri, cancellationToken).ConfigureAwait(false);
+    }
+
+    public async Task<PipelineUpdate> GetUpdate(string pipelineId, string updateId, CancellationToken cancellationToken = default)
+    {
+        var requestUri = $"{ApiVersion}/pipelines/{pipelineId}/updates/{updateId}";
+        var response = await HttpGet<Dictionary<string, PipelineUpdate>>(this.HttpClient, requestUri, cancellationToken).ConfigureAwait(false);
+        return response["update"];
+    }
+
+    public async Task<PipelineUpdatesList> ListUpdates(
+        string pipelineId,
+        string pageToken = null,
+        int? maxResults = null,
+        string untilUpdateId = null,
+        CancellationToken cancellationToken = default)
+    {
+        var requestUriSb = new StringBuilder($"{ApiVersion}/pipelines/{pipelineId}/updates");
+            
+        if (pageToken != null || maxResults != null || untilUpdateId != null)
+        {
+            requestUriSb.Append('?');
+        }
+
+        if (pageToken != null)
+        {
+            requestUriSb.Append($"&{pageToken}");
+        }
+
+        if (maxResults != null)
+        {
+            requestUriSb.Append($"&{maxResults}");
+        }
+
+        if (untilUpdateId != null)
+        {
+            requestUriSb.Append($"&{untilUpdateId}");
+        }

```

Comment:
The query strings are formatted wrong. It should be in this format:
```csharp
$"{ApiVersion}/pipelines/{pipelineId}/updates/updates?max_results={maxResults}&page_token={pageToken}&until_update_id={untilUpdateId}"
```

Also, to simplify the logic, we can assign a default value for the "max_results" parameter to 25 (and make this parameter non-nullable), so that we can always append max_results first, then conditionally append the rest.